version: '3.10'

services:
  auth_app:
    image: auth_app
    restart: always
    volumes:
      - ./ayth_app:/code
    build:
      context: .
      dockerfile: Dockerfile
    env_file:
      - ./.env
    ports:
      - "5000:5000"
    depends_on:
      redis:
        condition: service_healthy

  auth_db:
    image: postgres:13
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: auth_db
    restart: always
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "su postgres", "&&", "pg_isready"]
      interval: 5s
      timeout: 5s
      retries: 10

  redis:
    image: 'redis:6.2.6'
    expose:
      - "6379"
    healthcheck:
      test: ["CMD-SHELL", "redis-cli", "ping"]
      interval: 5s
      timeout: 5s
      retries: 10
    depends_on:
      auth_db:
        condition: service_healthy

  tests:
    image: auth_app
    entrypoint: [ "python", "-m", "pytest", "-vv", "tests/" ]
    env_file:
      - ./.env
    depends_on:
      - auth_app
      - auth_db

volumes:
  postgres_data:
